### DISCUSSION TITLE
Partial templates for websites/books

### DISCUSSION THREAD

**Discussion by maelle** [quarto-dev/quarto-cli#9350] [Q&A] [answered]
### Description

:wave: here! I'm reading https://quarto.org/docs/journals/templates.html#html-partials and wondering how the same idea of partial templates applies to books/websites, as the templates I see are not HTML: https://github.com/quarto-dev/quarto-cli/tree/main/src/resources/projects/website/templates

Does anyone know an example of custom partial usage for books/websites? Thanks!

**Comment by mcanouil** [comment, marked as answer]
It would apply the same way as a regular HTML document.

1. `quarto create project book demo Demo`
2. Edit `index.qmd` to include a partial such as `toc.html`, for example to change the background colour.

````qmd
---
format:
  html:
    template-partials:
      - toc.html
---

# Preface {.unnumbered}

This is a Quarto book.

To learn more about Quarto books visit <https://quarto.org/docs/books>.
````


````html
<nav id="$idprefix$TOC" role="doc-toc" style="background-color: brown;">
  $if(toc-title)$
  <h2 id="$idprefix$toc-title">$toc-title$</h2>
  $endif$ 
  $table-of-contents$
</nav>
````

<img width="959" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/8896044/0386765e-963e-4fbf-b6da-bac3b80cdcae">

Same thing for the `title-block.html` partial:
<img width="1624" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/8896044/402fc3a4-1c7b-457b-ac78-20221cd5051f">

**Reply by maelle** [reply]
mille mercis !

**Comment by maelle** [comment]
Thank you! How do you establish which partials are used for a given page?

**Reply by mcanouil** [reply]
I believe/assume they are all used😅
So, setting them on the document frontmatter, should only you to make advanced customisation involving all the partials.

Note: I'am not entirely sure I have understood the question, especially what you mean by "page".

**Reply by maelle** [reply]
I thought that maybe websites/books used different partials, but your idea did work for my use case: https://github.com/ropensci-review-tools/babelquarto/pull/34

**Reply by cscheid** [reply]
> Thank you! How do you establish which partials are used for a given page?

This is a great question.

Unfortunately, it's Pandoc that activates (or doesn't) the partials for a given template, and as far as I can tell, there's no way for us to know (even after the fact). This would have to be a new Pandoc feature, and I don't know how it would work in the context of that system.

**Reply by maelle** [reply]
Thank you! Well I'm lucky it was simply metadata.html in my case :grin:
